### PR TITLE
create_delta_layer: improve misleading TODO comment

### PR DESCRIPTION
**Pull request by problame**
Context: https://github.com/neondatabase/neon/pull/4441#discussion_r1228086608

### CODE REVIEW

**Review by skyzh — APPROVED** [review]
> this with a single fsync in future refactors

I admit this is misleading... I intended to say "we can simply call virtual_file.fsync() instead of par_fsync" 🤣

File: pageserver/src/tenant/timeline.rs [hunk 1]
```diff
@@ -3018,16 +3018,18 @@ impl Timeline {
...
         // TODO: If we're running inside 'flush_frozen_layers' and there are multiple
-        // files to flush, it might be better to first write them all, and then fsync
-        // them all in parallel.
-
-        // First sync the delta layer. We still use par_fsync here to keep everything consistent. Feel free to replace
-        // this with a single fsync in future refactors.
+        // files to flush, the fsync overhead can be reduces as follows:
+        // 1. write them all to temporary file names
+        // 2. fsync them
+        // 3. rename to the final name
```
Comment (LizardWizzard) [inline pageserver/src/tenant/timeline.rs:3030]: Arent we doing it this way already? (I mean the rename)

**Review by LizardWizzard — COMMENTED** [review]
WDYT if we just call `fsync_path` from par_fsync?

> I intended to say "we can simply call virtual_file.fsync() instead of par_fsync"
+1

**Comment by problame** [comment]
> WDYT if we just call fsync_path from par_fsync?

You mean inline par_fsync by just calling `fsync_path()` on these `len()==1` slices?

**Pull request merged**